### ISSUE TITLE
stderr: '/bin/sh: 1: /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot: Permission denied'

### ISSUE THREAD

**Issue by ZerooCool** [fail2ban/fail2ban#3106]
Hi,
Any idea why i have " Permission denied " ?

sudo systemctl restart fail2ban && tail -f -n 30 /var/log/fail2ban.log

```
2021-09-22 11:38:03,891 fail2ban.utils          [51482]: ERROR   7feda161a230 -- exec: /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 66.249.65.218
2021-09-22 11:38:03,892 fail2ban.utils          [51482]: ERROR   7feda161a230 -- stderr: '/bin/sh: 1: /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot: Permission denied'
2021-09-22 11:38:03,892 fail2ban.utils          [51482]: ERROR   7feda161a230 -- returned 126
2021-09-22 11:38:03,904 fail2ban.utils          [51482]: ERROR   7feda161a230 -- exec: /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 66.249.65.220
2021-09-22 11:38:03,904 fail2ban.utils          [51482]: ERROR   7feda161a230 -- stderr: '/bin/sh: 1: /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot: Permission denied'
2021-09-22 11:38:03,905 fail2ban.utils          [51482]: ERROR   7feda161a230 -- returned 126
```

The script for apache-fakegooglebot : https://wiki.visionduweb.fr/index.php?title=Installer_et_utiliser_Fail2ban#R.C3.A8gle_apache-fakegooglebot

Right for folder and files :
jail.d : 
-rw-r--r-- 1 root root  205 19 sept. 03:55 apache-fakegooglebot.conf

filter.d : 
-rw-r--r-- 1 root root   325 21 sept. 19:19 apache-fakegooglebot.conf

drwxr-xr-x 2 root root  4096 22 sept. 11:36 ignorecommands
-rw-r--r-- 1 root root  1418 22 sept. 11:36 apache-fakegooglebot

OS : Debian Bullseye
Firewall : Iptables 1.8.7

**Comment by ZerooCool** [comment]
`sudo chmod u+x /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot`

I am surprised because on my new installation in a virtual machine, by default, the file is not executable.
Is this really the right approach? 

Then, this is resolved.

**Comment by sebres** [comment]
Fail2ban has its own ignore-command [apache-fakegooglebot](https://github.com/fail2ban/fail2ban/blob/master/config/filter.d/ignorecommands/apache-fakegooglebot)... why do you install it from somewhere?

> Any idea why i have " Permission denied " ?

Normally (in practice) root user has usually all possible capabilities, so should be able to execute file without `x` permission.
But since I don't know how it is configured on your system and whether fail2ban is running as root or possibly as some unprivileged user...

1. ensure to grant `x` permission:
```
?sudo? chmod +x /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot
```
2. check the shebang of this is correct (e. g. `#!/usr/bin/env fail2ban-python`) and points to the same python-version fail2ban is installed for (or to `fail2ban-python` link).
3. check whether you can start it directly from shell, for instance:
```
$ /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 192.0.2.1; echo $?
1
$ /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 66.249.66.1; echo $?
0
```

**Comment by ZerooCool** [comment]
I have make a copy paste, no ? It's not the same code ? I go look that in this repo.
`Fail2ban has its own ignore-command apache-fakegooglebot... why do you install it from somewhere?`

```
I had deleted the file, which had been recreated with 644 rights. Execute rights were missing.

Apply the executable user right with chmod u + x on the apache-fakegooglebot script in the ignorecommands folder to resolve the error.
Chmod 755 is the default chmod for the file and also resolves the error. chmod 744 would also work.

Resolved:
sudo chmod 755 /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 
```

```
/etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 66.249.66.1; echo $?
-bash: /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot: Permission non accordée
126

sudo /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 66.249.66.1; echo $?
0
```

**Comment by ZerooCool** [comment]
I have change the code for the script to my wiki, and use the last version from Github and Fail2ban project.
I mean, it's ok now.

**Comment by ZerooCool** [comment]
Disregard this issue.
I checked again and the file is executable on a new installation.

It was a mistake, because I deleted the file due to improper handling.
I recreated the file myself in 644, which caused the error.

The chmod mod applied on a fresh install appears to be 755 and if I understand correctly this is the correct value.
If this is the case, it is not necessary to investigate further, the error is corrected on my end.
